File: java/09_pgm_avion_interface/src/co/edu/campusucc/poo/implement/TipoComercial.java
package co.edu.campusucc.poo.implement;

import co.edu.campusucc.poo.Avion;
import co.edu.campusucc.poo.interfaces.IAvionStandar;

public class TipoComercial extends Avion implements IAvionStandar {

    public TipoComercial() {
        super();
        setTipo("Comercial");
    }

    @Override
    public void startingEngine() {
        System.out.println("⛽Check Fuel ✅");
        for (int i = 0; i < 3; i++) {
            System.out.println("💥💥Spark💥💥");
            try {
                Thread.sleep(1000 - (i * 200));
            } catch (InterruptedException e) {
                System.err.println("⛔:" + e);
            }
        }
        System.out.println("Started the Motor...✈️✈️");
    }

    @Override
    public void stopingEngine() {
        System.out.println("✈️✈️ Check Engine ✅");
        for (int i = 0; i < 2; i++) {
            System.out.println("⛔ Stoping Engine...⛔‼️");
            try {
                Thread.sleep(1000 - (i * 400));
            } catch (InterruptedException e) {
                System.err.println("⛔:" + e);
            }
        }
        System.out.println("Stoped...⛔");
    }

}

File: c++/09_pgm_avion_interface/src/co/edu/campusucc/poo/implement/TipoComercial.cpp
#include <iostream>
#include "Avion.h"

class TipoComercial : public Avion {
public:
    TipoComercial() {
        setTipo("Comercial");
    }

    void startingEngine() override {
        std::cout << "⛽Check Fuel ✅\n";
        for (int i = 0; i < 3; i++) {
            std::cout << "💥💥Spark💥💥\n";
            try {
                std::this_thread::sleep_for(std::chrono::milliseconds(1000 - (i * 200)));
            } catch (std::exception& e) {
                std::cerr << "⛔:" << e.what() << '\n';
            }
        }
        std::cout << "Started the Motor...✈️✈️\n";
    }

    void stopingEngine() override {
        std::cout << "✈️✈️ Check Engine ✅\n";
        for (int i = 0; i < 2; i++) {
            std::cout << "⛔ Stoping Engine...⛔‼️\n";
            try {
                std::this_thread::sleep_for(std::chrono::milliseconds(1000 - (i * 400)));
            } catch (std::exception& e) {
                std::cerr << "⛔:" << e.what() << '\n';
            }
        }
        std::cout << "Stoped...⛔\n";
    }
};
